scripts/summarize-changes.js

```diff
@@ -0,0 +1,363 @@
+        const lines = part.value.split('\n');
+
+        // The logic here is a bit complex - the purpose is to find the 'context' for a change,
+        // which is usually the name of the interface or property that the changed member belongs to.
+        if (!part.added && !part.removed) {
+             // Look for context in lines preceding a change block
+            lines.forEach(line => {
+                const match = line.match(contextNameRegex);
+                if (match && !changeBlockActive) { // Only update context if we are not inside a change block
+                    lastPotentialContextName = match[1];
+                }
+            });
+            changeBlockActive = false; // Reset flag when entering context block
+        } else {
+            // This part has additions or removals
+            hasVisibleChanges = true;
+            changeBlockActive = true; // Set flag when inside a change block
+            if (contextNameForChange === null && lastPotentialContextName !== null) {
+                 // Capture the last known context name *before* the first change line
+                 contextNameForChange = lastPotentialContextName;
+            }
+
+            const prefix = part.added ? '+ ' : '- ';
+            lines.forEach((line, idx) => {
+                 // Avoid adding trailing newline as a separate diff line if it exists
+                if (idx === lines.length - 1 && line === '') return;
+                 outputLines.push(`    ${prefix}${line}`);
+            });
+        }
+    });
+
+    // Close the diff block
+    outputLines.push('    ```');
+
+    // Ensure we only return a diff block if there were actual visible changes
+    return {
+         diffBlock: hasVisibleChanges ? outputLines : [],
+         contextName: contextNameForChange // Return the context identified before the first +/- line
+     };
+}
+
+/**
+ * Compares two sets of interface definitions and returns a summary string array.
+ * Operates on signatures assumed to be JSDoc-free due to pre-processing.
+ * @param {Map<string, {name: string, members: Map<string, {name: string, signature: string}>}>} oldInterfaces
+ * @param {Map<string, {name: string, members: Map<string, {name: string, signature: string}>}>} newInterfaces
+ * @returns {string[]}
+ */
+function compareInterfaces(oldInterfaces, newInterfaces) {
+    const changes = [];
+    const allNames = new Set([...oldInterfaces.keys(), ...newInterfaces.keys()]);
+
+    for (const name of allNames) {
+        const oldI = oldInterfaces.get(name);
+        const newI = newInterfaces.get(name);
+
+        // In the unlikely event that an entire interface was added or removed,
+        // Handle it here.
+        if (!oldI) {
+                 changes.push(`- **Added Interface:** \`${name}\``);
+            continue;
+            }
+
+        if (!newI) {
+                changes.push(`- **Removed Interface:** \`${name}\``);
+            continue;
+             }
+
```

Comment:
The indentation are off here (and some other part of this file). Is there something wrong with your formatter?

Reply:
It's because formatting is run through eslint, and that's set to ignore js files. I'll run prettier across it.